### PR TITLE
Ajout des fonctions de vérification de fin du jeu

### PR DESCRIPTION
**Pull request by m4laclypse**
Contient également des fonctions de test associées.

### CODE REVIEW

File: 2048game/end_game.py [hunk 1]
```diff
@@ -5,3 +5,25 @@ def is_grid_full(game_grid):
             if x==' ':
                 return False
     return True
+
+def is_game_over(game_grid) :
+    tempBool = True
+    for row in game_grid :
```
Comment (oteboul) [inline 2048game/end_game.py:11]: no space before `:`

Comment (oteboul) [inline 2048game/end_game.py:11]: i believe you can find a better way using the built in functions `any` and `map`.

File: 2048game/end_game.py [hunk 2]
```diff
@@ -5,3 +5,25 @@ def is_grid_full(game_grid):
...
+        tempBool = tempBool and not " " in row
```
Comment (oteboul) [inline 2048game/end_game.py:12]: better `" " not in row`. it s more pythonic

File: 2048game/end_game.py [hunk 3]
```diff
@@ -5,3 +5,25 @@ def is_grid_full(game_grid):
...
+    return tempBool
+    
+def get_row_tile_max(row) :
+    maxValue = 0
+    for x in row :
+        if x != " " :
```
Comment (oteboul) [inline 2048game/end_game.py:18]: what about turning the " " into 0 ??

**Review by oteboul — COMMENTED** [review]
some small comments. please address

File: 2048game/end_game.py (hunk 1 above)
Comment (m4laclypse) [reply 2048game/end_game.py:11]: Done using map and reduce.

File: 2048game/end_game.py (hunk 2 above)
Comment (m4laclypse) [reply 2048game/end_game.py:12]: Replaced with something else, no longer relevant.

File: 2048game/end_game.py (hunk 3 above)
Comment (m4laclypse) [reply 2048game/end_game.py:18]: It is unnecessary here considering we have a base value for maxValue of 0, we need not compare it to 0s

**Pull request merged**